sunpy/io/setup_package.py

```diff
@@ -9,7 +9,11 @@
 
 def get_extensions():
 
-    if get_compiler() == 'msvc' or os.environ.get("SUNPY_NO_BUILD_ANA_EXTENSION", None):
+    if (
+        get_compiler() == 'msvc' or
+        os.environ.get("SUNPY_NO_BUILD_ANA_EXTENSION", None) or
+        "/OpenAstronomy/build-python-dist/" in os.environ.get("GITHUB_ACTION_PATH", "")
```

Comment:
Is this because we can't pass through env vars to the action workflows?

Reply:
Unfortunately yes. It would be better if we could avoid GitHub Actions specific code inside the package

Reply:
Why can't we pass env vars through again?

Reply:
It's in the list of limitations and I have tried before and it doesn't get passed through. (They do get passed into actions though.) https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations

Reply:
Can we hack round that in the workflow? Add it as a custom var which gets parsed by our python script?

Reply:
Yes. We could add a step that runs this bash command for each line in a string of env vars. https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

Reply:
I've created an issue https://github.com/OpenAstronomy/github-actions-workflows/issues/52